### PR TITLE
Fix #10352

### LINKED ISSUE #10352: Generated c code is not compile with the vcc cl.exe before 2012 after v0.19
After the v0.19, the module init function contains code "{TFrame FR_; FR_.len = 0;}" before c variable define, and this is not allowed with classical c compiler.

This is code generated after v0.19:
```nim
N_LIB_PRIVATE N_NIMCALL(void, systemInit000)(void) {
{
	TFrame FR_; FR_.len = 0;
}
	int T1_;
	int T2_;
	int T3_;
nimRegisterThreadLocalMarker(TM_Q5wkpxktOdTGvlSRo9bzt9aw_17);
	genericReset((void*)(&strDesc_D0UzA4zsDu5tgpJQ9a9clXPg), (&NTI_TuG1vdYwChKtx4TOkRnAjw_));
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.size = ((NI) 4);
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.kind = ((tyEnum_TNimKind_jIBKr1ejBgsfM33Kxw4j7A) 28);
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.flags = 2;
	T1_ = (int)0;
	T1_ = _fileno(stdin);
	_setmode(T1_, _O_BINARY);
	T2_ = (int)0;
	T2_ = _fileno(stdout);
	_setmode(T2_, _O_BINARY);
	T3_ = (int)0;
	T3_ = _fileno(stderr);
	_setmode(T3_, _O_BINARY);
	initAllocator_IZ2oOmqdsYQ7vixQGKSDcAsystem();
	initStackBottom();
	initGC_njssp69aa7hvxte9bJ8uuDcg();
	registerSignalHandler_njssp69aa7hvxte9bJ8uuDcg_2();
}
``` 

And this is the code generate before v0.19
```nim
N_LIB_PRIVATE N_NIMCALL(void, systemInit000)(void) {
	int T1_;
	int T2_;
	int T3_;
nimRegisterThreadLocalMarker(TM_Q5wkpxktOdTGvlSRo9bzt9aw_17);
	genericReset((void*)(&strDesc_D0UzA4zsDu5tgpJQ9a9clXPg), (&NTI_qQQlSworN6w6Qnxtmq0s5Q_));
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.size = ((NI) 4);
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.kind = ((tyEnum_TNimKind_jIBKr1ejBgsfM33Kxw4j7A) 28);
	strDesc_D0UzA4zsDu5tgpJQ9a9clXPg.flags = 2;
	T1_ = (int)0;
	T1_ = _fileno(stdin);
	_setmode(T1_, _O_BINARY);
	T2_ = (int)0;
	T2_ = _fileno(stdout);
	_setmode(T2_, _O_BINARY);
	T3_ = (int)0;
	T3_ = _fileno(stderr);
	_setmode(T3_, _O_BINARY);
	initAllocator_IZ2oOmqdsYQ7vixQGKSDcAsystem();
	initStackBottom();
	initGC_njssp69aa7hvxte9bJ8uuDcg();
	registerSignalHandler_njssp69aa7hvxte9bJ8uuDcg_2();
}
```

### PR DESCRIPTION
**Pull request by slangmgh**

### CODE REVIEW

**Comment by slangmgh** [comment]
The #10415 doesn't fix the issue completely, so I reopen this pr.

**Comment by slangmgh** [comment]
This fix only add '{' and '}' around the code follow the 'Frame' code. 
The generated c code like:
```nim
N_LIB_PRIVATE N_NIMCALL(void, compiler_stock_dataInit000)(void) {
	tyObject_TimezonecolonObjectType__F8OvqlxXyGXRSiK9c1fCDVw* T1_;
{
	TFrame FR_; FR_.len = 0;
	EMPTY_DATE_29bPbNs1RkaQ5RRFFecB32w.Sup.m_type = (&NTI_d9cK9cyKZPJJxFXUmx8dSpPw_);
}
{
	nimfr_("stock_data", "stock_data.nim");
nimRegisterGlobalMarker(TM_MFQ41p5Rlwq6TW4ubcDw4g_2);
	nimln_(484, "stock_data.nim");
	T1_ = (tyObject_TimezonecolonObjectType__F8OvqlxXyGXRSiK9c1fCDVw*)0;
	T1_ = utc_iX2Ltm3Mj5gsgJ5hFxUKgQ();
	initDateTime_qf1NugU9bsahSTtx9aMSE83g(((NI) 1), ((tyEnum_Month_JS0JuFvCBbC2GL3xXBnPQw) 1), ((NI) 1970), ((NI) 12), ((NI) 0), ((NI) 0), T1_, (&EMPTY_DATE_29bPbNs1RkaQ5RRFFecB32w));
	popFrame();
}
}
```

File: compiler/cgen.nim [hunk 1]
```diff
@@ -1362,6 +1362,7 @@ proc genInitCode(m: BModule) =
       add(prc, genSectionEnd(cpsStmts, m.config))
     addf(prc, "}$N", [])
 
+  addf(prc, "{$N", [])
```
Comment (LemonBoy) [inline compiler/cgen.nim]: Minor nit, can you move lines 1330-1338 below this `addf`? This way all the `initProc` codegen is grouped together.

Comment (slangmgh) [reply compiler/cgen.nim]: ok, the lines are moved.

**Pull request merged**